component-catalog-connectors/build-a-custom-connector.md

```diff
@@ -1,3 +1,215 @@
+
+The common public properties are:
+ - "learn more" link, providing access to to connector documentation
+ - catalog instance name
+ - catalog instance description
+ - runtime type filter, which governs the kind of components the connector makes available
+ - component categories, which group the components together in the pipeline editor
+
+Connector schemas can optionally also include custom public properties. For illustrative purposes, the quickstart template connector defines a required custom property and an optional custom property. Required properties are typically used to collect catalog connectivity information, such as a server URL, or API token. 
+
+#### Customize the schema
+
+The quickstart connector default schema file `TODO-catalog.json` is located in the [`todo_catalog_connector`](/component-catalog-connectors/connector-template/todo_catalog_connector/) directory.
+1. Rename the file to reflect your catalog type.
+1. Using an IDE of your choice, find and replace all occurrences of '`TODO-catalog.json`' in all files in the `component-catalog-connectors/connector-template` directory. (There should be seven.)
+1. Open the renamed `TODO-catalog.json` schema file in an editor.
+1. To avoid breaking the connector, do not: 
+     - Remove existing properties, unless indicated otherwise in the comments
+     - Change existing property names, unless indicated otherwise in the comments
+     - Change existing property values, unless indicated otherwise in the comments
+1. Customize the following common catalog connector schema properties:
+   - `$ID`: replace the URL with a valid URL for the renamed schema file.
+   - `title`: Replace the placeholder value, as desired
+   - `name`: Replace the placeholder value `todo-catalog` with a unique identifier for your catalog connector type.
+   - `uihints.title`: Replace the placeholder value with the value of the `title` property.
+   - `uihints.reference_url`: This is the "Learn more" link mentioned earlier. Replace the placeholder as desired.
+   - `properties.schemaname.const`: Replace with the new value of `name`.
+   - `properties.metadata.properties.description.default`: This is the default value for the catalog instance description. Replace the placeholder as desired. (Empty value is okay.)  
+   - `properties.metadata.runtime_type.enum`: This list defines the runtime types that the connector supports. The quickstart connector is configured for Kubeflow Pipelines only. Specify `["KUBEFLOW_PIPELINES"]`, `["APACHE_AIRFLOW"]`, or `["KUBEFLOW_PIPELINES", "APACHE_AIRFLOW]`, as appropriate. 
+   - `properties.metadata.runtime_type.default`: The value defines the default selection for `properties.metadata.runtime_type.enum`. (Empty value is okay.)
+   - `properties.metadata.categories.default`: This list defines the categories to which the retrieved components will be assigned by default. The user can override the default selection in the connector's configuration screen.
```

Comment:
Should we include a default in the schema template? Currently there is not one for `categories`. Alternatively, maybe we could just add phrasing to indicate that a user can add a key/value pair for this default field if desired.

Reply:
Done! I had an empty default `[]` yesterday, which triggered
```
Validation failed for instance 'hello' using the todo-catalog schema with error: '[]' is not of type 'array'.
```


I've added a value and confirmed that it's working

Reply:
I thought that might have been where that came from 😄